tests/Message/FetchPaymentMethodRequestTest.php

```diff
@@ -234,4 +238,42 @@ public function testSendPayPalSuccess()
 
         $this->assertSame('https://soap.prodtest.sj.vindicia.com/18.0/PaymentMethod.wsdl', $this->getLastEndpoint());
     }
+
+    /**
+     * @return void
+     */
+    public function testSendApplePaySuccess()
+    {
+        $this->setMockSoapResponse('FetchApplePayPaymentMethodSuccess.xml', array(
+            'PAYMENT_METHOD_ID' => $this->paymentMethodId,
+            'PAYMENT_METHOD_REFERENCE' => $this->paymentMethodReference,
+            'PAYMENT_INSTRUMENT_NAME' => $this->paymentInstrumentName,
+            'PAYMENT_NETWORK' => $this->paymentNetwork,
+            'TRANSACTION_IDENTIFIER' => $this->transactionIdentifier,
+            'PAYMENT_DATA' => $this->paymentData,
+            'COUNTRY' => $this->card['country'],
+            'POSTCODE' => $this->card['postcode']
+        ));
+
+        $response = $this->request->send();
+
+        $this->assertTrue($response->isSuccessful());
+        $this->assertFalse($response->isRedirect());
+        $this->assertFalse($response->isPending());
+        $this->assertSame('OK', $response->getMessage());
+        
+        $paymentMethod = $response->getPaymentMethod();
+        $this->assertInstanceOf('\Omnipay\Vindicia\PaymentMethod', $paymentMethod);
+        $this->assertSame($this->paymentMethodId, $response->getPaymentMethodId());
+        $this->assertSame($this->paymentMethodReference, $response->getPaymentMethodReference());
+        $this->assertSame($this->paymentMethodId, $paymentMethod->getId());
+        $this->assertSame($this->paymentMethodReference, $paymentMethod->getReference());
+        $this->assertSame('ApplePay', $paymentMethod->getType());
+        $card = $paymentMethod->getCard();
+        $this->assertInstanceOf('\Omnipay\Common\CreditCard', $card);
+        $this->assertSame($this->card['country'], $card->getCountry());
+        $this->assertSame($this->card['postcode'], $card->getPostcode());
+
```

Comment:
There are no nameValues in the xml file, so I am not checking for them here like in paypal.